mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -241,6 +252,12 @@ object KMeansModel extends MLReadable[KMeansModel] {
       }
       val model = new KMeansModel(metadata.uid, new MLlibKMeansModel(clusterCenters))
       DefaultParamsReader.getAndSetParams(model, metadata)
+      DefaultParamsReader.loadInitialModel[KMeansModel](path, sc) match {
```

Comment:
minor: If we are going to use a pattern match here, let's use it in the `KMeansReader` too. I think either is fine, but let's use a single convention. Also, if you eliminate the pattern match entirely, then the imports for Success and Failure can be removed.